lib/rogue/models/coordinate.rb

```diff
@@ -5,94 +5,72 @@ def initialize(x, y)
 
   attr_reader :x, :y
 
-  CHASE_DIRECTIONS = {:above? => :down, :below? => :up, :left_of? => :right, :right_of? => :left, \
-  :topleft_of? => :bottomright, :topright_of? => :bottomleft, :bottomright_of? => :topleft, :bottomleft_of? => :topright}
-
-  def eql?(object)
-    object.x == x && object.y == y
-  end
-
-  def hash
-    [x, y].hash
-  end
-
-  def up
-    Coordinate.new(x, y-1)
-  end
-
-  def down
-    Coordinate.new(x, y+1)
-  end
-
-  def left
-    Coordinate.new(x-1, y)
-  end
-
-  def right
-    Coordinate.new(x+1, y)
-  end
-
-  def topleft
-    Coordinate.new(x-1, y-1)
-  end
-
-  def topright
-    Coordinate.new(x+1, y-1)
-  end
-
-  def bottomleft
-    Coordinate.new(x-1, y+1)
-  end
-
-  def bottomright
-    Coordinate.new(x+1, y+1)
+  DIRECTIONS = [:left, :topleft, :up, :topright, :right, :bottomright, :down, :bottomleft]
+  DIRECTIONS_DIAGONALS_FIRST = DIRECTIONS.partition { |dir| DIRECTIONS.index(dir).odd? }.flatten
+
+  ROTATE_PATTERN = [-1, -1, 0, +1, +1, +1, 0, -1]
+  X_ROTATION = ROTATE_PATTERN
+  Y_ROTATION = X_ROTATION.rotate(-2)
+
+  def method_missing(direction, *args)
+    if DIRECTIONS.include?(direction)
+      rotation_index = DIRECTIONS.index(direction)
+      Coordinate.new((x + X_ROTATION[rotation_index]), (y + Y_ROTATION[rotation_index]))
+    elsif DIRECTIONS.include?(unquery(direction))
+      check_in_direction(unquery(direction), args[0])
+    end
   end
 
   def adjacent?(coord)
-    [:up, :down, :left, :right, :topleft, :topright, :bottomleft, :bottomright].each do |direction|
+    DIRECTIONS.each do |direction|
       return true if self.send(direction).eql?(coord)
     end
     false
   end
 
-  def above?(coord)
-    self.y < coord.y
+  def direction_to_follow(coord)
```

Comment:
i think it'd actually be good to move this logic out into its own class, something which deals with AI movement. That way we can keep the coordinate class as a value object and treat the other class as a decision maker.